pkg/querier/queryrange/labels_cache.go

```diff
@@ -0,0 +1,99 @@
+package queryrange
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/go-kit/log"
+
+	"github.com/grafana/loki/pkg/querier/queryrange/queryrangebase"
+	"github.com/grafana/loki/pkg/storage/chunk/cache"
+	"github.com/grafana/loki/pkg/storage/chunk/cache/resultscache"
+)
+
+type cacheKeyLabels struct {
+	Limits
+	transformer UserIDTransformer
+}
+
+// GenerateCacheKey generates a cache key based on the userID, split duration and the interval of the request.
+// It also includes the label name and the provided query for label values request.
+func (i cacheKeyLabels) GenerateCacheKey(ctx context.Context, userID string, r resultscache.Request) string {
+	lr := r.(*LabelRequest)
```

Comment:
i think in both cases, the req cannot be anything else because of the way we wire things up. This gets plugged into lables tripperware.

I can add the assert, but it requires changing the interface and it's implementations to return bool in a few places. not a lot of work, but on the fence if we should do it

Reply:
OK, if you're certain then we can leave it as is.